contracts/feature-tests/composability/promises-features/src/call_promises.rs

```diff
@@ -1,24 +1,26 @@
 multiversx_sc::imports!();
 
-use crate::common::{self, CallbackData};
+use crate::{
+    common::{self, CallbackData},
+    vault_proxy,
+};
 
 #[multiversx_sc::module]
 pub trait CallPromisesModule: common::CommonModule {
-    #[proxy]
-    fn vault_proxy(&self) -> vault::Proxy<Self::Api>;
-
     #[endpoint]
     #[payable("*")]
     fn forward_promise_accept_funds(&self, to: ManagedAddress) {
         let payment = self.call_value().egld_or_single_esdt();
         let gas_limit = self.blockchain().get_gas_left() / 2;
-        self.vault_proxy()
-            .contract(to)
+
+        self.tx()
+            .to(&to)
+            .typed(vault_proxy::VaultProxy)
             .accept_funds()
-            .with_egld_or_single_esdt_transfer(payment)
             .with_gas_limit(gas_limit)
-            .async_call_promise()
-            .register_promise()
+            .with_egld_or_single_esdt_transfer(payment)
+            .async_call()
```

Comment:
No more need for `async_call()` it does nothing, it's just for backwards compatibility.